src/CommonTypes.jl

```diff
@@ -64,24 +68,237 @@ Base.broadcastable(x::LiquidType) = tuple(x)
     IceType
 
 The type for cloud ice condensate
-"""
-struct IceType <: AbstractCloudType end
+
+# Fields
+$(DocStringExtensions.FIELDS)
+"""
+struct IceType{FT} <: AbstractCloudType
+    "cloud ice crystals length scale"
+    r0::FT
+    "cloud ice mass size relation coefficient me"
+    me::FT
+    "cloud ice mass size relation coefficient delm"
+    Δm::FT
+    "cloud ice mass_size_relation_coefficient chim"
+    χm::FT
+    "density ice water"
+    ρ::FT
+    "cloud ice size distribution coefficient n0"
+    n0::FT
+    "m0"
+    m0::FT
+    "ice snow threshold radius"
+    r_ice_snow::FT
+end
+
+function IceType(::Type{FT}) where {FT}
+    toml_dict = CP.create_toml_dict(FT)
+    (; data) = toml_dict
+    r0 = FT(data["cloud_ice_crystals_length_scale"]["value"])
+    me = FT(data["cloud_ice_mass_size_relation_coefficient_me"]["value"])
+    Δm = FT(data["cloud_ice_mass_size_relation_coefficient_delm"]["value"])
+    χm = FT(data["cloud_ice_mass_size_relation_coefficient_chim"]["value"])
+    ρ = FT(data["density_ice_water"]["value"])
+    n0 = FT(data["cloud_ice_size_distribution_coefficient_n0"]["value"])
+    m0 = FT(4 / 3) * π * ρ * r0^me
+    r_ice_snow = FT(data["ice_snow_threshold_radius"]["value"])
+    return IceType(r0, me, Δm, χm, ρ, n0, m0, r_ice_snow)
```

Comment:
This is really great. I like how the type carries all the necessary parameters. I'm fine with how you split things, even if it results in an annoying interface for the 2-moment velocity. 

We can revisit this when we have the full P3 scheme done. Because then we will have two schemes that cover all four types (Liquid, Ice, Rain, Snow). Maybe then a different partitioning will be more elegant